src/connections/allContacts.js

```diff
@@ -1,11 +1,26 @@
 import get from 'lodash/get'
 
+import uniqWith from 'lodash/uniqWith'
+import isEqual from 'lodash/isEqual'
 import { mergeContact } from '../helpers/mergeContact'
-import { findContactsWithSamePhoneOrEmail } from '../helpers/findContact'
 import { DOCTYPE_CONTACTS } from '../helpers/doctypes'
+import { buildContactsQueryByEmailAdressOrPhoneNumber } from '../queries/queries'
 
 export const importContact = async (client, attributes) => {
-  const contacts = await findContactsWithSamePhoneOrEmail(attributes)(client)
+  const addresses = (attributes['email'] || []).map(email => email.address)
+  const numbers = (attributes['phone'] || []).map(phone => phone.number)
+
+  const contactQueryByEmailAdressOrPhoneNumber = buildContactsQueryByEmailAdressOrPhoneNumber(
+    addresses,
+    numbers
+  )
+
+  const result = await client.query(
+    contactQueryByEmailAdressOrPhoneNumber.definition
+  )
+
+  const contacts = uniqWith(result.data, isEqual)
```

Comment:
pourquoi ce "fix" ici ? 🤔 (on sort d'ailleurs du scope de refactoring comme annoncé dans le commit...)

Reply:
I tried to explain into the description of my commit. The check was previously done with an generic function. I chose to simplify it by making it specific. The objective is to be able to understand it quickly and optimize it in the future.